### DISCUSSION TITLE
Réglage tête thermostatique Sonoff TRVZB

### DISCUSSION THREAD

**Discussion by Masha76490** [haade-administrator/haade.fr#60] [Q&A]
Bonjour, je reprend le message d'aide d'un autre utilisateur sur le forum de hacf. Nous cherchons tous les deux de l'aide sur un même problème. Voici le message : 

"Bonjour,

J’ai installé des têtes thermostatiques Sonoff TRVZB

Pour le moment, si je règle une consigne, par exemple à 18 °C, elle va s’ouvrir pour monter en température, se fermer une fois arrivée à la consigne, et attendre d’être redescendue de 1 °C en dessous de la consigne (donc 17 °C) avant de s’ouvrir à nouveau pour chauffer.

Je me demandais s’il était possible de modifier le delta de température pour qu’elle remette le radiateur en chauffe à partir de -0,5 °C en dessous de la consigne, au lieu de -1 °C. Avec le fonctionnement actuel, la température varie beaucoup dans la pièce, et ce n’est pas très confortable.

Merci pour votre aide."

En vous remerciant aussi pour votre aide.

**Comment by haade-administrator** [comment]
Bonjour, déjà je suppose que tu es sous home assistant, si oui utilises-tu z2m ou zha ?
Pour commencer fait la mise à jour 1.2.1 il y a deux-trois news
le module dispose d'un mode off/auto/heat, pour ne pas utiliser la consigne il faut être sur heat il me semble
teste voir et reviens vers moi, on peut faire un petit script pour améliorer ça, cependant une consigne de -1 ne devrait pas se ressentir ou alors tu as un tirage d'air froid constant quelque part.

**Reply by Masha76490** [reply]
Bonjour,
Alors oui je suis sous HA, et j'ulise Z2m. Et je suis déjà sous 1.2.1 pour les têtes thermostats. 
Comme dans votre article, pour ne pas utiliser le thermostat, faux, interne à la tête, j'utilise TRV Calibrator pour contourner le systeme. Le problème que je rencontre, c'est que je si je demande 18 degres. La tête va bien monter à 18 mais elle va attendre d'être à 17 avant de se réenclencher alors que j'aimerais qu'elle se réenclenche à 17.5 minimum. (le système off/auto/heat fonctionne bien)
Je sais pas si je suis clair et si j'ai répondu correctement a votre message.

**Reply by haade-administrator** [reply]
Oui j'ai parfaitement compris, à part modifier le firmware tu ne pourras modifier la consigne à moins 1, mais tu peux automatiser un shuntant le système comme suivant
déclencheur:
si local_temperature < occupied setpoint

condition

quand occupied_setpoint - local_température > 0.5

alors 

passer occupied_hetaing _setpoint à 19.5

attendre ( calcule le temps de montée en température à 19.5 ) et rajoute à la condition attendre

attendre 10 min

passer occupied..... à 19°C

**Reply by Masha76490** [reply]
Je pensais avoir résolu avec Better Thermostat le problème mais finalement non.
Je débute totalement sur HA et je ne comprends pas la démarche que vous me décrivez. Est il possible d’avoir + d’explications svp

**Reply by haade-administrator** [reply]
Le sonoff ne possède pas de fonction de réglage de la température de consigne -1 ou -2° du coup rien ne pourras être envisagé avec divers modules pour cette fonction.
Par contre la sonde du thermostat est erroné quand le radiateur chauffe, elle va capter la chaleur au lieu de capter la température ambiante de la pièce du coup suivant l'inertie de tes radiateurs cette température va mettre plus ou moins de temps à baisser au niveau de la sonde.
Je te conseil dans un premier temps d'acheter un thermostat externe:

**itead:** [snzb-02p](https://itead.cc/product/sonoff-zigbee-temperature-and-humidity-sensor-snzb-02p/ref/122/)
**domadoo:** [snzb-02p](https://www.domadoo.fr/fr/chauffage-clim/6732-sonoff-capteur-de-temperature-et-d-humidite-zigbee-30-avec-support-snzb-02p.html?domid=39)

car une consigne de 1°c ne devrait pas être ressentie dans la pièce

si tu as un thermomètre mesure voir la température quand la pièce se refroidit et compare par rapport à la température du thermostat
